From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Micah Hausler <mhausler@amazon.com>
Date: Sun, 1 Nov 2020 09:24:08 -0800
Subject: [PATCH] --EKS-PATCH-- Use GNU date

Description:
* Use GNU date if available for builds on darwin.

Upstream PR, Issue, KEP, etc. links:
* Based on unmerged and closed Kubernetes PR #113210 (https://github.com/kubernetes/kubernetes/pull/100044). See below
for additional information.

If this patch is based on an upstream commit, how (if at all) do this patch and the upstream source differ?
* There are minor differences between this patch and the upstream implementation of this feature. When this patch was
created, it was taken from the proposed changes in PR #100044 (https://github.com/kubernetes/kubernetes/pull/100044).
Due to inactivity, the PR closed without being merged and left a couple of comments unanswered. Later, PR #113210
(https://github.com/kubernetes/kubernetes/pull/113210) was opened to implement the same feature, but it included
additional changes aimed to address the open questions from the previous PR. This PR was approved and merged by
upstream.

If this patch's changes have not been added by upstream, why not?
* N/A

Other patches related to this patch:
* None

Changes made to this patch after its initial creation and reasons for these changes:
* None

Kubernetes version this patch can be dropped:
* v1.26 -- upstream will include this feature with https://github.com/kubernetes/kubernetes/pull/113210)

Signed-off-by: Jyoti Mahapatra<jyotima@amazon.com>
---
 hack/lib/version.sh | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/hack/lib/version.sh b/hack/lib/version.sh
index 3d0646b1c59..98df9dbbef1 100644
--- a/hack/lib/version.sh
+++ b/hack/lib/version.sh
@@ -163,8 +163,14 @@ kube::version::ldflags() {
       "-X 'k8s.io/component-base/version.${key}=${val}'"
     )
   }
+  DATE=date
+  if which gdate &>/dev/null; then
+      DATE=gdate
+  elif which gnudate &>/dev/null; then
+      DATE=gnudate
+  fi
 
-  add_ldflag "buildDate" "$(date ${SOURCE_DATE_EPOCH:+"--date=@${SOURCE_DATE_EPOCH}"} -u +'%Y-%m-%dT%H:%M:%SZ')"
+  add_ldflag "buildDate" "$(${DATE} ${SOURCE_DATE_EPOCH:+"--date=@${SOURCE_DATE_EPOCH}"} -u +'%Y-%m-%dT%H:%M:%SZ')"
   if [[ -n ${KUBE_GIT_COMMIT-} ]]; then
     add_ldflag "gitCommit" "${KUBE_GIT_COMMIT}"
     add_ldflag "gitTreeState" "${KUBE_GIT_TREE_STATE}"
